client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,9 +12,67 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { getUnconnectedSite } from 'state/selectors';
 import { updateJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
+export const announceRequestFailure = ( { dispatch } ) =>
+	dispatch(
+		errorNotice( translate( 'Could not fetch settings from site. Please try again later.' ) )
+	);
+
+const receiveJetpackOnboardingSettings = (
+	{ dispatch },
+	action,
+	{ data: { onboarding: settings } }
+) => {
+	const { siteId } = action;
+
+	if ( ! settings ) {
+		return announceRequestFailure( { dispatch }, action );
+	}
```

Comment:
we can provide `fromApi` and this can be handled separately and leave this receiver more segregated in its responsibility

```js
fromApi: response => {
	if ( ! response.onboarding ) {
		throw new Error( 'missing onboarding settings' )
	}

	return response.onboarding;
}
```

in this case if the `onboarding` property is missing then Calypso will call `onFailure` for you with that `Error` passed in as the `error` parameter. otherwise it will call this receiver but instead of the full response data it will get the `settings` only

Reply:
TIL. Awesome! 133bf5c